### ISSUE TITLE
Whois function for the ircclient class

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3133]
|<img alt="Jem777's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Jem777 reported|
|-|-|
|Trac ID|trac#3133|
|Type|enhancement|
|Created|2008-03-28 09:37:42Z|
|Branch|https://github.com/twisted/twisted/tree/irc-whois-3133|

In the irc.py there is a function whois for the server protocol, but not for the client

<details><summary>Searchable metadata</summary>

```
trac-id__3133 3133
type__enhancement enhancement
reporter__Jem777 Jem777
priority__normal normal
milestone__ 
branch__branches_irc_whois_3133 branches/irc-whois-3133
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1206697062000000 1206697062000000
changetime__1222292859000000 1222292859000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24720]) Branching to 'irc-whois-3133'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Do @since tags apply to classes only or methods as well? New functionality like ```whois``` could use an @since i think.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Yep, `@since` tags should apply to any new API.  I'll add one to `whois`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [24771]) @since: 8.2

refs [#3133](https://github.com/twisted/twisted/issues/3133)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner|
|-|-|

How's that?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

Docs look good, not sure about the code, I'll let others decide on that.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to @exarkun|
|-|-|

Cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [24922]) Merge irc-whois-3133

Author: exarkun
Reviewer: thijs, therve
Fixes: [#3133](https://github.com/twisted/twisted/issues/3133)

Add `IRCClient.whois` for looking up information about IRC users.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
